### PR TITLE
Added extraction of reactions from a message object

### PR DESCRIPTION
**Pull request by KatiGithub**

### CODE REVIEW

**Review by dfireBird — CHANGES_REQUESTED** [review]
Thanks for opening the PR. Can you reset the changes made on `package-lock.json`.

File: src/utils/index.ts [hunk 1]
```diff
@@ -13,3 +16,29 @@ export function extractEmoji(message: string): Array<string> {
...
+    const reactionsManager = message.reactions;
+
+    let objreactions: any = {};
+
+    for (var reaction_constructor of reactionsManager.cache) {
+        let ObjReactionMessage = reaction_constructor[1];
+        let emoji = ObjReactionMessage.emoji;
+
+        if (emoji.toString() in objreactions) {
+            objreactions[emoji.toString()] = objreactions[emoji.toString()] + 1;
```
Comment (dfireBird) [inline src/utils/index.ts]: Instead of using emoji's name as key instead we can use the id as key.

File: src/utils/index.ts [hunk 2]
```diff
@@ -13,3 +16,29 @@ export function extractEmoji(message: string): Array<string> {
...
+// 
+// Return an object with those reactions.
+// *If need be, change emoji.toString() to emoji.toId() to access the ID of the emojis.*
+
+export function extractReaction(message: Message) {
+    const reactionsManager = message.reactions;
+
+    let objreactions: any = {};
+
+    for (var reaction_constructor of reactionsManager.cache) {
```
Comment (dfireBird) [inline src/utils/index.ts:32]: Why not use `let` here?

**Comment by KatiGithub** [comment]
> Thanks for opening the PR. Can you reset the changes made on `package-lock.json`.

I'm new here, not sure as to how to change the package-lock.json back.

**Comment by dfireBird** [comment]
> > Thanks for opening the PR. Can you reset the changes made on `package-lock.json`.
> 
> I'm new here, not sure as to how to change the package-lock.json back.

Use `git rm package-lock.json` commit that and push it.

File: src/utils/index.ts (hunk 1 above)
Comment (KatiGithub) [reply src/utils/index.ts]: The id might be null so the only other viable option is emoji.identifer.

Comment (dfireBird) [reply src/utils/index.ts]: The id being null is very rare case from what I researched. Even discord uses that id for most purposes. So I think it's better to use them. It's also what the bot uses to store the emojis in DB.

**Comment by dfireBird** [comment]
Thanks for making changes I mentioned one last change I want is extract only emojis that belong to guild the message is sent, now you're extracting all emojis.

**Comment by KatiGithub** [comment]
> Thanks for making changes I mentioned one last change I want is extract only emojis that belong to guild the message is sent, now you're extracting all emojis.

Can you clarify how I'm "extracting all emojis"? I only passed into the function one message parameter which is supposed to only take the reaction from one  message.

**Review by dfireBird — APPROVED** [review]

**Comment by dfireBird** [comment]
> Can you clarify how I'm "extracting all emojis"? I only passed into the function one message parameter which is supposed to only take the reaction from one message.

I can make the suggested changes myself!

Thanks for your hard work! I'll be now merge this PR!

**Pull request merged**